Client/Frontend/Browser/BrowserViewController/BVCHelper.swift

```diff
@@ -0,0 +1,222 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+import Shared
+import Data
+import BraveShared
+
+class BVCHelper {
+  enum SpecialSchemeType {
+    case aboutScheme, bookmarklet
+    case appleMapsURL(URL)
+    case storeURL(URL)
+    case externalAppScheme(URL)
+    case braveScheme(URL)
+    case internalLink
+    case websiteRedirect(URLRequest)
+    case buyVPN
+  }
+  
+  static func handleSpecialActions(for navigationAction: WKNavigationAction) -> SpecialSchemeType? {
```

Comment:
Can we please try to avoid creating a class with static functions and try to add this helper as a dependency to wherever it is needed to be used ?